homeassistant/components/switcher_kis/switch.py

```diff
@@ -183,3 +204,117 @@ async def async_turn_on_with_timer_service(self, timer_minutes: int) -> None:
         await self._async_call_api(API_CONTROL_DEVICE, Command.ON, timer_minutes)
         self.control_result = True
         self.async_write_ha_state()
+
+
+class SwitchereShutterChildLockBaseSwitchEntity(SwitcherEntity, SwitchEntity):
+    """Representation of a Switcher shutter base switch entity."""
+
+    _attr_device_class = SwitchDeviceClass.SWITCH
+    _attr_entity_category = EntityCategory.CONFIG
+    _attr_icon = "mdi:lock-open"
+    _cover_id: int
+
+    def __init__(self, coordinator: SwitcherDataUpdateCoordinator) -> None:
+        """Initialize the entity."""
+        super().__init__(coordinator)
+        self.control_result: bool | None = None
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """When device updates, clear control result that overrides state."""
+        self.control_result = None
+        self.async_write_ha_state()
+
+    async def _async_call_api(self, api: str, *args: Any) -> None:
+        """Call Switcher API."""
+        _LOGGER.debug("Calling api for %s, api: '%s', args: %s", self.name, api, args)
+        response: SwitcherBaseResponse | None = None
+        error = None
+
+        try:
+            async with SwitcherApi(
+                self.coordinator.data.device_type,
+                self.coordinator.data.ip_address,
+                self.coordinator.data.device_id,
+                self.coordinator.data.device_key,
+                self.coordinator.token,
+            ) as swapi:
+                response = await getattr(swapi, api)(*args)
+        except (TimeoutError, OSError, RuntimeError) as err:
+            error = repr(err)
+
+        if error or not response or not response.successful:
+            _LOGGER.error(
+                "Call api for %s failed, api: '%s', args: %s, response/error: %s",
+                self.coordinator.name,
+                api,
+                args,
+                response or error,
+            )
+            self.coordinator.last_update_success = False
```

Comment:
This method repeats twice in this file and has 2 more occurrences in `light` and `cover` platforms. Please move it to the base entity (`entity.py`) and remove it from all other places.

Reply:
It is also affects the tests